staging/src/k8s.io/client-go/util/cert/cert.go

```diff
@@ -87,6 +87,26 @@ func NewSelfSignedCACert(cfg Config, key crypto.Signer) (*x509.Certificate, erro
 	return x509.ParseCertificate(certDERBytes)
 }
 
+// NewCSR creates a new CSR
+func NewCSR(cfg Config, key crypto.Signer) (*x509.CertificateRequest, error) {
```

Comment:
the certificates packages in client-go are kind of a kitchen sink and we are working on removing a lot of the surface. @awly would you keep these functions here? I'm wondering if adding this to kubeadm utilities would be preferable.

Reply:
I'd move this to pkiutil.
Everything under client-go is intended for clients of k8s that do not want to vendor the entire k8s codebase. We should keep only things that are relevant for clients here.

Reply:
Ack. Everything removed from staging.